src/Requester.ts

```diff
@@ -0,0 +1,21 @@
+import * as https from 'https';
+import { TimeoutError } from './errors';
+
+export interface IRequester {
+    request<T>(url: string ): Promise<T>;
+}
+
+export class Requester implements IRequester {
+    public request<T>(url: string): Promise<T> {
+        return new Promise((resolve, reject) => {
+            const req = https.get(url, res => {
+                let body = '';
+                res.on('data', str => body = body + str.toString());
+                res.on('end', () => resolve(JSON.parse(body)));
```

Comment:
Particularly since the response is being occluded in this promise, you probably want to reject on a non-2xx error code.